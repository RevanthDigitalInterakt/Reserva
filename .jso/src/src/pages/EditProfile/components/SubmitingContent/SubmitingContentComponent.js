  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _react = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[2]);
  function SubmitingContentComponent(_ref) {
    var isRegister = _ref.isRegister,
      handleSubmitForm = _ref.handleSubmitForm,
      formEditIsValid = _ref.formEditIsValid;
    var navigation = (0, _$$_REQUIRE(_dependencyMap[3]).useNavigation)();
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsxs)(_$$_REQUIRE(_dependencyMap[4]).Fragment, {
      children: [isRegister && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[5]).Box, {
        mb: "nano",
        justifyContent: "space-between",
        flexDirection: "row",
        zIndex: 2,
        children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[5]).Box, {
          paddingLeft: "nano",
          mt: "sm",
          width: "100%",
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[6]).Button, {
            testID: "com.usereserva:id/submitingcontent_button_submit_register",
            title: "SALVAR",
            variant: "primarioEstreito",
            inline: true,
            onPress: handleSubmitForm,
            disabled: formEditIsValid
          })
        })
      }), !isRegister && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsxs)(_$$_REQUIRE(_dependencyMap[5]).Box, {
        flex: 1,
        width: _reactNative.Dimensions.get('window').width,
        justifyContent: "space-between",
        paddingX: "xxxs",
        alignItems: "center",
        flexDirection: "row",
        height: 85,
        style: {
          elevation: _$$_REQUIRE(_dependencyMap[7]).platformType.ANDROID === 'android' ? 10 : 0,
          position: 'absolute',
          zIndex: 999,
          bottom: 42.5,
          left: 0
        },
        backgroundColor: "white",
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[5]).Box, {
          width: 0.45454545454545453,
          paddingRight: "nano",
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[6]).Button, {
            title: "CANCELAR",
            testID: "com.usereserva:id/submitingcontent_button_go_back_no_register",
            variant: "primarioEstreitoOutline",
            inline: true,
            onPress: function onPress() {
              return navigation.goBack();
            }
          })
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[5]).Box, {
          width: 0.45454545454545453,
          paddingLeft: "nano",
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[4]).jsx)(_$$_REQUIRE(_dependencyMap[6]).Button, {
            title: "SALVAR",
            variant: "primarioEstreito",
            testID: "com.usereserva:id/submitingcontent_button_submit_no_register",
            inline: true,
            onPress: handleSubmitForm,
            disabled: formEditIsValid
          })
        })]
      })]
    });
  }
  var _default = exports.default = SubmitingContentComponent;
