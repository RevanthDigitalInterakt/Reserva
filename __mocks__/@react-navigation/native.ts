const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
