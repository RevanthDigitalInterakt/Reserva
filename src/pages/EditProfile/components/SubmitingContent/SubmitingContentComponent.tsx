import React from 'react';
import { Box, Button } from '@usereservaapp/reserva-ui';
import { Dimensions } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { platformType } from '../../../../utils/platformType';

interface ISubmitingContentComponent {
  isRegister: boolean
  handleSubmitForm: () => void
  formEditIsValid: boolean
}

function SubmitingContentComponent({
  isRegister,
  handleSubmitForm,
  formEditIsValid,
}: ISubmitingContentComponent): JSX.Element {
  const navigation = useNavigation();

  return (
    <>
      {isRegister && (
        <Box
          mb="nano"
          justifyContent="space-between"
          flexDirection="row"
          zIndex={2}
        >
          <Box paddingLeft="nano" mt="sm" width="100%">
            <Button
              testID="com.usereserva:id/submitingcontent_button_submit_register"
              title="SALVAR"
              variant="primarioEstreito"
              inline
              onPress={handleSubmitForm}
              disabled={formEditIsValid}
            />
          </Box>
        </Box>
      )}

      {!isRegister && (
        <Box
          flex={1}
          width={Dimensions.get('window').width}
          justifyContent="space-between"
          paddingX="xxxs"
          alignItems="center"
          flexDirection="row"
          height={85}
          style={{
            elevation: platformType.ANDROID === 'android' ? 10 : 0, position: 'absolute', zIndex: 999, bottom: 85 / 2, left: 0,
          }}
          backgroundColor="white"
        >
          <Box width={1 / 2.2} paddingRight="nano">
            <Button
              title="CANCELAR"
              testID="com.usereserva:id/submitingcontent_button_go_back_no_register"
              variant="primarioEstreitoOutline"
              inline
              onPress={() => navigation.goBack()}
            />
          </Box>

          <Box width={1 / 2.2} paddingLeft="nano">
            <Button
              title="SALVAR"
              variant="primarioEstreito"
              testID="com.usereserva:id/submitingcontent_button_submit_no_register"
              inline
              onPress={handleSubmitForm}
              disabled={formEditIsValid}
            />
          </Box>
        </Box>
      )}
    </>
  );
}

export default SubmitingContentComponent;
