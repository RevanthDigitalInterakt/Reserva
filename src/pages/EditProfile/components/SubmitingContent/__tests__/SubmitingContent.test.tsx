import React from 'react';
import { theme } from '@usereservaapp/reserva-ui';
import { ThemeProvider } from 'styled-components/native';
import {
  act, fireEvent, render, screen,
} from '@testing-library/react-native';
import SubmitingContentComponent from '../SubmitingContentComponent';

const SubmitingContentProps = {
  handleSubmitForm: jest.fn(),
  formEditIsValid: false,
};

describe('Submiting Content', () => {
  it('renders without error and match snapshot', () => {
    render((
      <ThemeProvider theme={theme}>
        <SubmitingContentComponent isRegister={false} {...SubmitingContentProps} />
      </ThemeProvider>
    ));
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

describe('Testing isRegister=true', () => {
  it('check visible go back button', async () => {
    render((
      <ThemeProvider theme={theme}>
        <SubmitingContentComponent isRegister {...SubmitingContentProps} />
      </ThemeProvider>
    ));
    expect(screen.getByTestId('com.usereserva:id/submitingcontent_button_submit_register')).toBeVisible();
  });

  it('check pressable submit form', async () => {
    render((
      <ThemeProvider theme={theme}>
        <SubmitingContentComponent isRegister {...SubmitingContentProps} />
      </ThemeProvider>
    ));

    await act(async () => {
      await fireEvent.press(screen.getByTestId('com.usereserva:id/submitingcontent_button_submit_register'));
    });

    expect(SubmitingContentProps.handleSubmitForm).toHaveBeenCalledTimes(1);
  });
});

describe('Testing isRegister=false', () => {
  it('check visible go back button', async () => {
    render((
      <ThemeProvider theme={theme}>
        <SubmitingContentComponent isRegister={false} {...SubmitingContentProps} />
      </ThemeProvider>
    ));
    expect(screen.getByTestId('com.usereserva:id/submitingcontent_button_go_back_no_register')).toBeVisible();
  });

  it('check visible save button', () => {
    render((
      <ThemeProvider theme={theme}>
        <SubmitingContentComponent isRegister={false} {...SubmitingContentProps} />
      </ThemeProvider>
    ));
    expect(screen.getByTestId('com.usereserva:id/submitingcontent_button_submit_no_register')).toBeVisible();
  });
});

describe('SubmitingContentTest', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('check pressable submit form', async () => {
    render((
      <ThemeProvider theme={theme}>
        <SubmitingContentComponent isRegister={false} {...SubmitingContentProps} />
      </ThemeProvider>
    ));

    await act(async () => {
      await fireEvent.press(screen.getByTestId('com.usereserva:id/submitingcontent_button_submit_no_register'));
    });

    expect(SubmitingContentProps.handleSubmitForm).toHaveBeenCalledTimes(1);
  });
});
