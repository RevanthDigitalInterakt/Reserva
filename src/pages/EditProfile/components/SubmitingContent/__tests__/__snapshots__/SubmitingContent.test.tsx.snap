// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submiting Content renders without error and match snapshot 1`] = `
<View
  alignItems="center"
  backgroundColor="white"
  flex={1}
  flexDirection="row"
  height={85}
  justifyContent="space-between"
  paddingX="xxxs"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 85,
        "justifyContent": "space-between",
        "paddingLeft": 16,
        "paddingRight": 16,
        "width": 750,
      },
      Object {
        "bottom": 42.5,
        "elevation": 10,
        "left": 0,
        "position": "absolute",
        "zIndex": 999,
      },
    ]
  }
  width={750}
>
  <View
    paddingRight="nano"
    style={
      Array [
        Object {
          "paddingRight": 8,
          "width": "45.45454545454545%",
        },
      ]
    }
    width={0.45454545454545453}
  >
    <View
      alignItems="stretch"
      style={
        Array [
          Object {
            "alignItems": "stretch",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#333333",
            "borderWidth": 1,
            "color": "#333333",
            "fontFamily": "Nunito-Bold",
            "fontSize": 13,
            "justifyContent": "center",
            "minWidth": 130,
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 16,
          }
        }
        testID="submitingcontent_button_go_back_no_register"
      >
        <View
          alignItems="center"
          flexDirection="row"
          justifyContent="center"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            justifyContent="center"
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              color="preto"
              fontFamily="nunitoBold"
              fontSize={13}
              letterSpacing={1.6}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Nunito-Bold",
                    "fontSize": 13,
                    "letterSpacing": 1.6,
                  },
                ]
              }
            >
              CANCELAR
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    paddingLeft="nano"
    style={
      Array [
        Object {
          "paddingLeft": 8,
          "width": "45.45454545454545%",
        },
      ]
    }
    width={0.45454545454545453}
  >
    <View
      alignItems="stretch"
      style={
        Array [
          Object {
            "alignItems": "stretch",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#333333",
            "color": "#FFF",
            "fontFamily": "Nunito-Bold",
            "fontSize": 13,
            "justifyContent": "center",
            "minWidth": 130,
            "opacity": 1,
            "paddingBottom": 16,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 16,
          }
        }
        testID="submitingcontent_button_submit_no_register"
      >
        <View
          alignItems="center"
          flexDirection="row"
          justifyContent="center"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            justifyContent="center"
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              color="white"
              fontFamily="nunitoBold"
              fontSize={13}
              letterSpacing={1.6}
              style={
                Array [
                  Object {
                    "color": "#FFF",
                    "fontFamily": "Nunito-Bold",
                    "fontSize": 13,
                    "letterSpacing": 1.6,
                  },
                ]
              }
            >
              SALVAR
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
